### PR TITLE
refactor: refactor curp_server storage implementation

### PR DESCRIPTION
**Pull request by Phoenix500526**
Please briefly answer these questions:
make curp storage engine configurable.

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: curp/src/server/storage/db.rs [hunk 1]
```diff
@@ -13,15 +14,15 @@ const VOTE_FOR: &[u8] = b"VoteFor";
 const CF: &str = "curp";
 
 /// `RocksDB` storage implementation
```
Comment (iGxnon) [inline curp/src/server/storage/db.rs:16]: Should we modify this comment?

File: curp/src/server/storage/mod.rs [hunk 2]
```diff
@@ -35,4 +35,4 @@ pub(super) trait StorageApi: Send + Sync {
 }
 
 /// `RocksDB` storage implementation
```
Comment (iGxnon) [inline curp/src/server/storage/mod.rs:37]: Same as above

File: curp/src/server/storage/db.rs (hunk 1 above)
Comment (Phoenix500526) [reply curp/src/server/storage/db.rs:16]: Done

File: curp/src/server/storage/mod.rs (hunk 2 above)
Comment (Phoenix500526) [reply curp/src/server/storage/mod.rs:37]: Done

File: xline/src/main.rs [hunk 3]
```diff
@@ -267,6 +267,19 @@ struct ServerArgs {
...
+            "memory" => (StorageConfig::Memory, StorageConfig::Memory),
+            "rocksdb" => (
+                StorageConfig::RocksDB(args.data_dir.clone()),
+                StorageConfig::RocksDB(args.curp_dir.unwrap_or_else(|| {
+                    let mut path = args.data_dir;
+                    path.push("curp");
+                    path
+                })),
+            ),
+            &_ => unreachable!(),
```
Comment (themanforfree) [inline xline/src/main.rs:280]: Please print some useful information here, users may pass in other string in the command line

File: curp-test-utils/src/lib.rs [hunk 4]
```diff
@@ -1,8 +1,5 @@
 #![allow(dead_code, unused)]
```
Comment (themanforfree) [inline curp-test-utils/src/lib.rs:1]: I think we didn't need this, please remove it and remove dead codes

File: xline/src/main.rs (hunk 3 above)
Comment (Phoenix500526) [reply xline/src/main.rs:280]: Done

File: curp-test-utils/src/lib.rs (hunk 4 above)
Comment (Phoenix500526) [reply curp-test-utils/src/lib.rs:1]: Done

**Review by iGxnon — APPROVED** [review]

**Review by themanforfree — APPROVED** [review]

**Pull request merged**